drivers/Aqara/aqara-presence-sensor/src/fp2/device_manager.lua

```diff
@@ -0,0 +1,272 @@
+  device:emit_event(PresenceSensor.presence(evt_action))
+end
+
+function device_manager.movement_handler(driver, device, zone, evt_value)
+  if not device:supports_capability(PresenceSensor) then return end
+
+  local val = evt_value
+  local no_movement = function()
+    if not device:supports_capability(PresenceSensor) then return end
+    device:emit_event(MovementSensor.movement("inactive"))
+  end
+  device:set_field(MOVEMENT_TIMER, device.thread:call_with_delay(MOVEMENT_TIME, no_movement))
+
+  if val == 0 then
+    device:emit_event(MovementSensor.movement("entering"))
+  elseif val == 1 then
+    device:emit_event(MovementSensor.movement("leaving"))
+  elseif val == 2 then
+    device:emit_event(MovementSensor.movement("enteringLeft"))
+  elseif val == 3 then
+    device:emit_event(MovementSensor.movement("leavingRight"))
+  elseif val == 4 then
+    device:emit_event(MovementSensor.movement("enteringRight"))
+  elseif val == 5 then
+    device:emit_event(MovementSensor.movement("leavingLeft"))
+  elseif val == 6 then
+    device:emit_event(MovementSensor.movement("approaching"))
+  elseif val == 7 then
+    device:emit_event(MovementSensor.movement("movingAway"))
+  end
+end
+
+function device_manager.zone_presence_handler(driver, device, zone, evt_value)
+  if not device:supports_capability(PresenceSensor) then return end
+
+  local zoneInfo = multipleZonePresence.findZoneById(zone)
+  if not zoneInfo then
+    multipleZonePresence.createZone("zone" .. zone, zone)
+  end
+  local evt_action = multipleZonePresence.notPresent
+  if evt_value == 1 then evt_action = multipleZonePresence.present end
+  multipleZonePresence.changeState(zone, evt_action)
+  multipleZonePresence.updateAttribute(driver, device)
+end
+
+function device_manager.illuminance_handler(driver, device, zone, evt_value)
+  device:emit_event(capabilities.illuminanceMeasurement.illuminance(evt_value))
+end
+
+function device_manager.work_mode_handler(driver, device, zone, evt_value)
+  local cur_mode = device:get_latest_state(COMP_MODE, DeviceMode.ID, DeviceMode.mode.NAME) or 0
+  local mode = 1
+  local profile_name = "aqara-fp2-zoneDetection"
+  if not cur_mode then
+    device:emit_component_event(device.profile.components[COMP_MODE], DeviceMode.mode(FP2_MODES[mode]))
+    return
+  elseif evt_value == 0x03 then
+    if cur_mode == FP2_MODES[1] then return end
+  elseif evt_value == 0x05 then
+    if cur_mode == FP2_MODES[2] then return end
+    mode = 2
+    profile_name = "aqara-fp2-fallDetection"
+  elseif evt_value == 0x09 then
+    if cur_mode == FP2_MODES[3] then return end
+    mode = 3
+    profile_name = "aqara-fp2-sleepMonitoring"
+  end
+  device:emit_component_event(device.profile.components[COMP_MODE], DeviceMode.mode(FP2_MODES[mode]))
+  device:try_update_metadata({ profile = profile_name })
+end
+
+function device_manager.zone_quantities_handler(driver, device, zone, evt_value)
+  if not device:supports_capability(PresenceSensor) then return end
```

Comment:
I think this should be `multipleZonePresence`

Reply:
Okay. I'll fix it.